frontend/src/components/Layout/Carousel/Item/ItemsCarouselTitle.vue

```diff
@@ -31,96 +31,79 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
 import { BaseItemDto } from '@jellyfin/sdk/lib/generated-client';
 import { getLogo, ImageUrlInfo } from '@/utils/images';
 import { getItemDetailsLink } from '@/utils/items';
 
-export default defineComponent({
-  props: {
-    item: {
-      type: Object as () => BaseItemDto,
-      required: true
-    }
-  },
-  data() {
-    return {
-      itemLink: '',
-      titleString: '',
-      logoLink: '',
-      subtitle: ''
-    };
-  },
-  computed: {
-    logo(): ImageUrlInfo {
-      return getLogo(this.item);
-    }
-  },
-  watch: {
-    item: {
-      immediate: true,
-      handler(): void {
-        switch (this.item.Type) {
-          case 'MusicAlbum': {
-            if (this.item.AlbumArtists?.length) {
-              this.logoLink = getItemDetailsLink(
-                this.item.AlbumArtists[0],
-                'MusicArtist'
-              );
-            }
+const props = defineProps<{ item: BaseItemDto }>();
 
-            if (this.item.AlbumArtist) {
-              this.titleString = this.item.AlbumArtist;
-            }
+const { t } = useI18n();
 
-            if (this.item.Name) {
-              this.subtitle = this.item.Name;
-            }
+const itemLink = ref('');
```

Comment:
All of this can probably be computed instead of a watcher

Reply:
Definitely, I initially just copied the current behavior, but I'm happy to switch this. This could either be broken into multiple simple computed values, or a single computed value which returns an object. I'm going to go ahead with a single object since the logic for computing them all is related, but let me know if that's bad practice.